cloud/deployment/workerqueue/workerqueue.go

```diff
@@ -35,264 +35,329 @@ var (
+	if !(requestedWorkerQueue.MaxWorkerCount >= int(defaultOptions.MaxWorkers.Floor)) ||
+		!(requestedWorkerQueue.MaxWorkerCount <= int(defaultOptions.MaxWorkers.Ceiling)) {
+		errorMessage = fmt.Sprintf("max worker count must be between %d and %d", int(defaultOptions.MaxWorkers.Floor), int(defaultOptions.MaxWorkers.Ceiling))
 		return fmt.Errorf("%w: %s", errInvalidWorkerQueueOption, errorMessage)
 	}
 	// The floor for worker concurrency for hosted deployments is always 1 for all astro machines
 	workerConcurrenyFloor := 1
 	if !(requestedWorkerQueue.WorkerConcurrency >= workerConcurrenyFloor) ||
-		!(requestedWorkerQueue.WorkerConcurrency <= machineOptions.ConcurrentTasksMax) {
-		errorMessage = fmt.Sprintf("worker concurrency must be between %d and %d", workerConcurrenyFloor, machineOptions.ConcurrentTasksMax)
+		!(requestedWorkerQueue.WorkerConcurrency <= int(machineOptions.Concurrency.Ceiling)) {
+		fmt.Println("hosted function called")
+		errorMessage = fmt.Sprintf("worker concurrency must be between %d and %d", workerConcurrenyFloor, int(machineOptions.Concurrency.Ceiling))
 		return fmt.Errorf("%w: %s", errInvalidWorkerQueueOption, errorMessage)
 	}
-	if requestedWorkerQueue.PodCPU != "" {
-		return fmt.Errorf("%s %w %s", deployment.CeleryExecutor, ErrNotSupported, podCPUErrorMessage)
-	}
-	if requestedWorkerQueue.PodRAM != "" {
-		return fmt.Errorf("%s %w %s", deployment.CeleryExecutor, ErrNotSupported, podRAMErrorMessage)
-	}
 	return nil
 }
 
 // IsKubernetesWorkerQueueInputValid checks if the requestedQueue has all the necessary properties
 // required to create a worker queue for the KubernetesExecutor.
 // errNotSupported is returned for any invalid properties.
-func IsKubernetesWorkerQueueInputValid(requestedWorkerQueue *astro.WorkerQueue) error {
+func IsKubernetesWorkerQueueInputValid(queueToCreateOrUpdateHybrid *astroplatformcore.HybridWorkerQueueRequest) error {
 	var errorMessage string
-	if requestedWorkerQueue.Name != defaultQueueName {
+
+	if queueToCreateOrUpdateHybrid.Name != defaultQueueName {
 		errorMessage = "a non default worker queue in the request. Rename the queue to default"
 		return fmt.Errorf("%s %w %s", deployment.KubeExecutor, ErrNotSupported, errorMessage)
 	}
-	if requestedWorkerQueue.PodCPU != "" {
-		errorMessage = "pod cpu in the request. It will be calculated based on the requested worker type"
-		return fmt.Errorf("%s %w %s", deployment.KubeExecutor, ErrNotSupported, errorMessage)
-	}
-	if requestedWorkerQueue.PodRAM != "" {
-		errorMessage = "pod ram in the request. It will be calculated based on the requested worker type"
-		return fmt.Errorf("%s %w %s", deployment.KubeExecutor, ErrNotSupported, errorMessage)
-	}
-	if requestedWorkerQueue.MinWorkerCount != -1 {
+	if queueToCreateOrUpdateHybrid.MinWorkerCount != -1 {
 		errorMessage = "minimum worker count in the request. It can only be used with CeleryExecutor"
```

Comment:
seems this will never happen since we always set it to -1 in the call before so is this necessary? would this ever be hit?

Reply:
This function used to be used in more places

Reply:
I removed this check